src/explore-education-statistics-admin/src/pages/release/edit-release/content/ReleaseContext.tsx

```diff
@@ -0,0 +1,212 @@
+        }
+      } else if (draft.release?.[sectionKey]) {
+        const matchingSection = draft.release[sectionKey];
+        if (matchingSection) {
+          if (Array.isArray(matchingSection.content)) {
+            matchingSection.content.push(block);
+          } else {
+            matchingSection.content = [block];
+          }
+        }
+      }
+      return draft;
+    }
+    case 'UPDATE_SECTION_CONTENT': {
+      const { sectionContent, meta } = action.payload;
+      const { sectionId, sectionKey } = meta;
+      if (!draft.release?.[sectionKey]) {
+        throw new Error('ADD_BLOCK_TO_SECTION: failed');
+      }
+
+      if (sectionKey === 'content') {
+        const matchingSection = draft.release[sectionKey].find(
+          section => section.id === sectionId,
+        );
+        if (matchingSection) matchingSection.content = sectionContent;
+      } else if (draft.release?.[sectionKey]) {
+        const matchingSection = draft.release[sectionKey];
+        if (matchingSection) matchingSection.content = sectionContent;
+      }
+      return draft;
+    }
+    case 'ADD_CONTENT_SECTION': {
+      const { section } = action.payload;
+      if (draft.release) draft.release.content.push(section);
+      return draft;
+    }
+    case 'SET_CONTENT': {
+      const { content } = action.payload;
+      if (draft.release) {
+        draft.release.content = content;
+      }
+      return draft;
+    }
+    case 'UPDATE_CONTENT_SECTION': {
+      const { section, meta } = action.payload;
+      const { sectionId } = meta;
+
+      if (draft.release) {
+        const sectionIndex = draft.release.content.findIndex(
+          accordionSection => accordionSection.id === sectionId,
+        );
+        if (sectionIndex !== -1) draft.release.content[sectionIndex] = section;
+      }
+      return draft;
+    }
+    default: {
+      return draft;
+    }
+  }
+};
+
+function ReleaseProvider({ children }: ReleaseProviderProps) {
+  const [state, dispatch] = useImmerReducer(releaseReducer, {
+    release: undefined,
+    canUpdateRelease: false,
+    availableDataBlocks: [],
+    unresolvedComments: [],
+  });
+  return (
+    <ReleaseStateContext.Provider value={state}>
+      <ReleaseDispatchContext.Provider value={dispatch}>
+        {children}
+      </ReleaseDispatchContext.Provider>
+    </ReleaseStateContext.Provider>
+  );
+}
+
+function useReleaseState() {
+  const context = useContext(ReleaseStateContext);
+  if (context === undefined) {
+    throw new Error('useReleaseState must be used within a ReleaseProvider');
+  }
+  return context;
+}
+
+function useReleaseDispatch() {
+  const context = useContext(ReleaseDispatchContext);
+  if (context === undefined) {
+    throw new Error('useReleaseDispatch must be used within a ReleaseProvider');
+  }
+  return context;
+}
+
+export {
+  ReleaseProvider,
+  useReleaseState,
+  useReleaseDispatch,
+  // eslint-disable-next-line no-undef
+  ReleaseContextState,
```

Comment:
Put the exports directly on the types. The reason you're getting lint errors is because types/interfaces don't actually exist in Javascript land (hence being undefined) and consequently can't actually be exported within objects.